jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -0,0 +1,286 @@
+      APP_ROOT.resolve(JavaEntrypointConstructor.DEFAULT_RELATIVE_CLASSES_PATH_ON_IMAGE);
+
+  /** Absolute path of dependencies on container. */
+  private static final AbsoluteUnixPath DEPENDENCIES_PATH =
+      APP_ROOT.resolve(JavaEntrypointConstructor.DEFAULT_RELATIVE_DEPENDENCIES_PATH_ON_IMAGE);
+
+  /** Absolute path of additional classpath files on container. */
+  private static final AbsoluteUnixPath OTHERS_PATH = APP_ROOT.resolve("other");
+
+  /**
+   * Creates a new {@link JavaContainerBuilder} that uses distroless java as the base image. For
+   * more information on {@code gcr.io/distroless/java}, see <a
+   * href="https://github.com/GoogleContainerTools/distroless">the distroless repository</a>.
+   *
+   * @return a new {@link JavaContainerBuilder}
+   * @throws InvalidImageReferenceException if creating the base image reference fails
+   * @see <a href="https://github.com/GoogleContainerTools/distroless">The distroless repository</a>
+   */
+  public static JavaContainerBuilder fromDistroless() throws InvalidImageReferenceException {
+    return from(RegistryImage.named("gcr.io/distroless/java"));
+  }
+
+  /**
+   * Creates a new {@link JavaContainerBuilder} with the specified base image reference.
+   *
+   * @param baseImageReference the base image reference
+   * @return a new {@link JavaContainerBuilder}
+   * @throws InvalidImageReferenceException if {@code baseImageReference} is invalid
+   */
+  public static JavaContainerBuilder from(String baseImageReference)
+      throws InvalidImageReferenceException {
+    return from(RegistryImage.named(baseImageReference));
+  }
+
+  /**
+   * Creates a new {@link JavaContainerBuilder} with the specified base image reference.
+   *
+   * @param baseImageReference the base image reference
+   * @return a new {@link JavaContainerBuilder}
+   */
+  public static JavaContainerBuilder from(ImageReference baseImageReference) {
+    return from(RegistryImage.named(baseImageReference));
+  }
+
+  /**
+   * Creates a new {@link JavaContainerBuilder} with the specified base image reference.
+   *
+   * @param registryImage the {@link RegistryImage} that defines base container registry and
+   *     credentials
+   * @return a new {@link JavaContainerBuilder}
+   */
+  public static JavaContainerBuilder from(RegistryImage registryImage) {
+    return new JavaContainerBuilder(Jib.from(registryImage));
+  }
+
+  private final JibContainerBuilder jibContainerBuilder;
+  private final JavaLayerConfigurations.Builder layerConfigurationsBuilder =
+      JavaLayerConfigurations.builder();
+  private final List<String> jvmFlags = new ArrayList<>();
+  private final LinkedHashSet<String> classpath = new LinkedHashSet<>(4);
+
+  @Nullable private String mainClass;
+
+  private JavaContainerBuilder(JibContainerBuilder jibContainerBuilder) {
+    this.jibContainerBuilder = jibContainerBuilder;
+  }
+
+  /**
+   * Adds dependency JARs to {@code /app/libs} on the image.
```

Comment:
Perhaps we might not want to indicate where the location is on the image since if we change the constants we would have to update this javadoc and it might be an implementation detail that is not necessary to describe in this `JavaContainerBuilder` API.